src/core/core/rpc_engine.cpp

```diff
@@ -154,29 +154,16 @@ namespace dsn {
             // server address side effect
             auto req = call->get_request();
             auto sp = task_spec::get(req->local_rpc_code);
-            if (reply->from_address != req->to_address)
+            // TODO(qinzuoyan): because current rpc forward is fake (resend from client), we cannot
```

Comment:
I c, the faked forwarding makes this wrong. In the current case, however, ```ERR_OK == err``` is removed, which is wrong from my perspective. I'll try to fix together in next pr.